vibrato/src/trainer.rs

```diff
@@ -403,11 +396,254 @@ impl Trainer {
+    ///
+    /// # Arguments
+    ///
+    /// * `left_wtr` - Write sink targetting `left.def`.
+    /// * `right_wtr` - Write sink targetting `right.def`.
+    ///
+    /// # Errors
+    ///
+    /// [`VibratoError`](crate::errors::VibratoError) is returned when:
+    ///
+    /// - the merging weights fails, or
+    /// - the writing fails.
+    pub fn write_used_features<L, R>(&mut self, left_wtr: L, right_wtr: R) -> Result<()>
+    where
+        L: Write,
+        R: Write,
+    {
+        if self.merged_model.is_none() {
+            self.merged_model = Some(self.data.raw_model.merge()?);
+        }
+        let merged_model = self.merged_model.as_ref().unwrap();
+
+        // left
+        let mut left_wtr = BufWriter::new(left_wtr);
+        let mut left_features: Vec<_> = self
+            .data
+            .config
+            .feature_extractor
+            .left_feature_ids()
+            .iter()
+            .collect();
+        left_features.sort_unstable_by_key(|(_, v)| **v);
+        let feature_list = &merged_model.left_conn_to_right_feats;
+        for (conn_id, feat_ids) in feature_list[..feature_list.len() - 1].iter().enumerate() {
+            write!(&mut left_wtr, "{}", conn_id + 1)?;
+            for (i, feat_id) in feat_ids.iter().enumerate() {
+                if let Some(feat_id) = feat_id {
+                    let feat_str = &left_features[usize::from_u32(feat_id.get()) - 1].0;
+                    write!(&mut left_wtr, " {i}:{feat_str}")?;
+                }
+            }
+            writeln!(&mut left_wtr)?;
+        }
+
+        // right
+        let mut right_wtr = BufWriter::new(right_wtr);
+        let mut right_features: Vec<_> = self
+            .data
+            .config
+            .feature_extractor
+            .right_feature_ids()
+            .iter()
+            .collect();
+        right_features.sort_unstable_by_key(|(_, v)| **v);
+        let feature_list = &merged_model.right_conn_to_left_feats;
+        for (conn_id, feat_ids) in feature_list[..feature_list.len() - 1].iter().enumerate() {
+            write!(&mut right_wtr, "{}", conn_id + 1)?;
+            for (i, feat_id) in feat_ids.iter().enumerate() {
+                if let Some(feat_id) = feat_id {
+                    let feat_str = &right_features[usize::from_u32(feat_id.get()) - 1].0;
+                    write!(&mut right_wtr, " {i}:{feat_str}")?;
+                }
+            }
+            writeln!(&mut right_wtr)?;
+        }
+        Ok(())
+    }
+
+    /// Write the dictionary.
+    ///
+    /// # Arguments
+    ///
+    /// * `lexicon_wtr` - Write sink targetting `lex.csv`.
+    /// * `connector_wtr` - Write sink targetting `matrix.def`.
+    /// * `unk_handler_wtr` - Write sink targetting `unk.def`.
+    /// * `user_lexicon_wtr` - Write sink targetting `user.csv`. If no user-defined lexicon file is
+    ///                        specified, this function generates an empty file.
```

Comment:
```suggestion
    /// * `user_lexicon_wtr` - Write sink targetting `user.csv`. Set a dummy argument if no user-defined
    ///                        lexicon file is specified.
```
It would be better to clearly explain what the user should do.